### PR TITLE
assaf-java-s2i-update

### PR DESCRIPTION
**Pull request by amoldavsky**
**What this PR does / why we need it**:
Allows more flexibility around the S2I builds by allowing the
source image to be specified as a Docker build argument.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
You can either build the same way as before, OR you can do custom...
```
bash bin/build.sh \
	--build-source-image seldonio/core-builder:0.15 \
	--runtime-source-image adoptopenjdk/openjdk13:alpine-slim \
	jw-0.2.0-jre-13
```

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @amoldavsky. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/SeldonIO/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **adriangonz**
You can assign the PR to them by writing `/assign @adriangonz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: incubating/wrappers/s2i/java/Dockerfile.build [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM seldonio/core-builder:0.3
+ARG IMAGE_SOURCE=seldonio/core-builder:0.3
```
Comment (adriangonz) [inline incubating/wrappers/s2i/java/Dockerfile.build:1]: Could you try bumping the `core-builder` image to `0.16` (which is the latest)?

File: incubating/wrappers/s2i/java/Dockerfile.runtime [hunk 2]
```diff
@@ -1,4 +1,5 @@
-FROM openjdk:8u131-jre-alpine
+ARG IMAGE_SOURCE=openjdk:8u131-jre-alpine
```
Comment (adriangonz) [inline incubating/wrappers/s2i/java/Dockerfile.runtime:1]: Similarly as above, could you also bump the runtime base to `openjdk:13.0.1-jdk-buster` (which is what the latest `core-builder` uses)?

**Comment by adriangonz** [comment]
/approve

Nice one @amoldavsky ! Thanks for adding those changes in. I think this is good to go now.

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[adriangonz](https://github.com/SeldonIO/seldon-core/pull/2178#issuecomment-685409682 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-685409682/SeldonIO/seldon-core/blob/master/OWNERS)~~ [adriangonz]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**